modules/WFCORE-3684_JDK Module Support.adoc

```diff
@@ -0,0 +1,189 @@
+= Java Platform Module Support
+:author:            David Lloyd
+:email:             david.lloyd@redhat.com
+:toc:               left
+:icons:             font
+:keywords:          jigsaw,jpms,java9,jboss-modules
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+In Java 9, the platform classes are divided up into modules; the core set being:
+
+.List of core Java 9 modules
+[cols="3*m"]
+|======
+|java.base|java.compiler|java.datatransfer
+|java.desktop|java.instrument|java.jnlp
+|java.logging|java.management|java.management.rmi
+|java.naming|java.prefs|java.rmi
+|java.scripting|java.security.jgss|java.security.sasl
+|java.smartcardio|java.sql|java.sql.rowset
+|java.xml|java.xml.crypto|
+|======
+
+In addition, the ``java.se`` module re-exports many of these.
+
+As a first step towards alignment, we need to
+introduce the ability for modules to create module dependencies on
+these module names.  However, unless we want to require Java 9 or later from
+now on, the names must also work on Java 8.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFCORE-3705[WFCORE-3705]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/MODULES-254[MODULES-254]
+* https://issues.jboss.org/browse/WFBUILD-37[WFBUILD-37]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:mjurc@redhat.com[Michal Jurc]
+
+=== Affected Projects or Components
+
+* JBoss Modules
+* Selected WildFly Core modules
+* WildFly Build Tools
+
+=== Other Interested Projects
+
+* WildFly Swarm
+
+== Requirements
+
+=== Hard Requirements
+
+* Support all standard OpenJDK 9.x platform module names as built-in modules
```

Comment:
Should we really look into Java 9 and 10 with this RFE? Or just focus on Java 8 + Java 11 ea builds?

Reply:
I don't think it hurts to keep track of all the versions.  Also, it is important for compatibility (not just for users, but for us as well) to support as many of the existing Java 8 APIs (which we don't already replace) as possible, so this means aligning more closely with Java 9 than Java 11 (which is missing some of these APIs and introduces some modules which are not present in 8).

Of course once the container runs on Java 9 or higher, it's the JDK that provides the modules and not us.